clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/HostLevelIngestionStats.java

```diff
@@ -279,29 +279,36 @@ public HostLevelIngestionStats(
     this.totalHybridRecordsConsumedByRegionId =
         Collections.unmodifiableList(Arrays.asList(tmpTotalHybridRecordsConsumedByRegionId));
 
-    // Stats which are per-store only:
-
-    this.diskQuotaSensor =
-        registerSensor("global_store_disk_quota_allowed", new Gauge(() -> diskQuotaAllowedGauge), new Max());
-
+    // Register an aggregate metric for disk_usage_in_bytes metric
+    final boolean isTotalStats = isTotalStats();
     registerSensor(
         "disk_usage_in_bytes",
         new Gauge(
             () -> ingestionTaskMap.values()
                 .stream()
-                .filter(task -> task.getStoreName().equals(storeName))
-                .mapToLong(task -> task.getStorageEngine().getStoreSizeInBytes())
+                .filter(task -> isTotalStats ? true : task.getStoreName().equals(storeName))
+                .mapToLong(
+                    task -> isTotalStats
+                        ? task.getStorageEngine().getCachedStoreSizeInBytes()
+                        : task.getStorageEngine().getStoreSizeInBytes())
                 .sum()));
-
+    // Register an aggregate metric for rmd_disk_usage_in_bytes metric
     registerSensor(
         "rmd_disk_usage_in_bytes",
         new Gauge(
             () -> ingestionTaskMap.values()
                 .stream()
-                .filter(task -> task.getStoreName().equals(storeName))
-                .mapToLong(task -> task.getStorageEngine().getRMDSizeInBytes())
+                .filter(task -> isTotalStats ? true : task.getStoreName().equals(storeName))
+                .mapToLong(
+                    task -> isTotalStats
+                        ? task.getStorageEngine().getCachedRMDSizeInBytes()
+                        : task.getStorageEngine().getRMDSizeInBytes())
                 .sum()));
 
+    // Stats which are per-store only:
+    this.diskQuotaSensor =
+        registerSensor("global_store_disk_quota_allowed", new Gauge(() -> diskQuotaAllowedGauge), new Max());
```

Comment:
I see that this is moved code, not new code, but I am wondering whether it makes sense to have both a `Gauge` and a `Max` (or anything else, really) on the same sensor... the `Gauge` doesn't do anything when the sensor gets recorded, and `Max` is only relevant if the sensor gets recorded... I guess it "works" but it's a bit confusing, and maybe not what the author of that code expected...?

Reply:
Yeah, it is confusing, and I don't know the original intentions..